source/extensions/filters/http/cache/cache_headers_utils.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "envoy/common/time.h"
+#include "envoy/http/header_map.h"
+
+#include "absl/strings/string_view.h"
+#include "absl/time/time.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+using OptionalDuration = absl::optional<SystemTime::duration>;
+
+struct RequestCacheControl {
+  RequestCacheControl(bool must_validate = false, bool no_store = false, bool no_transform = false,
```

Comment:
I think the code would be simpler if RequestCacheControl and ResponseCacheControl's only constructors were the string_view ones. In tests, they can be constructed from string literals (e.g. "public, max-age=34").

Reply:
In tests, I am checking that a string cache-control header is parsed correctly to booleans and durations, how can I test this if I use a string literals?

Reply:
I could create a `TestRequestCacheControl` struct in the test file that inherits `RequestCacheControl` and defines that constructor. Same for `ResponseCacheControl`. What do you think?

Reply:
This would require adding a default constructor to `RequestCacheControl` and `ResponseCacheControl` though, it doesn't have to do anything but it must be defined. Also, I would not be able to initialize the members in the initializer list of the derived test classes, but rather in the constructor itself. It will be something like this:
```
struct TestRequestCacheControl : public RequestCacheControl {
  TestRequestCacheControl(bool must_validate, bool no_store, bool no_transform, bool only_if_cached,
                          OptionalDuration max_age, OptionalDuration min_fresh,
                          OptionalDuration max_stale) {
    must_validate_ = must_validate;
    no_store_ = no_store;
    no_transform_ = no_transform;
    only_if_cached_ = only_if_cached;
    max_age_ = max_age;
    min_fresh_ = min_fresh;
    max_stale_ = max_stale;
  }
};
```

Reply:
That works, and avoids complicating the public interface. (Ignore the string literal suggestion--I was erroneously thinking about tests for code that uses these structs.)